File: Daily Test/C++/Encloser Character Matrix 5.4.2025.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main() {
    int r, c, k;
    cin >> r >> c;
    vector<vector<string>> m(r, vector<string>(c));
    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            cin >> m[i][j];
    cin >> k;

    int a = r + 2 * k, b = c + 2 * k;
    vector<vector<string>> g(a, vector<string>(b, ""));

    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            g[i + k][j + k] = m[i][j];

    for (int i = 0; i < a; i += k) {
        for (int j = 0; j < b; j += k) {
            string bv = ((i / k + j / k) % 2 == 0) ? "1" : "0";
            for (int x = i; x < min(i + k, a); x++) {
                for (int y = j; y < min(j + k, b); y++) {
                    if (g[x][y] == "") g[x][y] = bv;
                }
            }
        }
    }

    for (int i = 0; i < a; i++) {
        for (int j = 0; j < b; j++)
            cout << g[i][j];
        cout << endl;
    }
    return 0;
}

File: Daily Test/Java/Encloser Character Matrix 5.4.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int r = sc.nextInt(), c = sc.nextInt();
        String[][] m = new String[r][c];
        for (int i = 0; i < r; i++)
            for (int j = 0; j < c; j++)
                m[i][j] = sc.next();
        int k = sc.nextInt();
        int a = r + 2 * k, b = c + 2 * k;
        String[][] g = new String[a][b];

        for (int i = 0; i < r; i++)
            for (int j = 0; j < c; j++)
                g[i + k][j + k] = m[i][j];

        for (int i = 0; i < a; i += k) {
            for (int j = 0; j < b; j += k) {
                String bv = ((i / k + j / k) % 2 == 0) ? "1" : "0";
                for (int x = i; x < Math.min(i + k, a); x++) {
                    for (int y = j; y < Math.min(j + k, b); y++) {
                        if (g[x][y] == null)
                            g[x][y] = bv;
                    }
                }
            }
        }

        for (int i = 0; i < a; i++) {
            for (int j = 0; j < b; j++)
                System.out.print(g[i][j]);
            System.out.println();
        }
    }
}